File: Clases 2018/Clases-C#/Simulacro_parcial_2_2018 (creacion de bug e historico en transaccion)/C#_TPS_BugTracker/TPS_BugTracker/BusinessLayer/Services/BugService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using TPS_BugTracker.BusinessLayer.Entities;
using TPS_BugTracker.DataLayer.DAOs;

namespace TPS_BugTracker.BusinessLayer.Services
{
    public class BugService
    {
        private BugDao oBugDao;
        public BugService()
        {
            oBugDao = new BugDao();
                   }
        public IList<Bug> consultarBugsConFiltros(List<object> @params)
        {
            return oBugDao.getBugByFilters(@params);
        }

        public Bug consultarBugsPorId(int id)
        {
            return oBugDao.getBugById(id);
        }

        public bool actualizarBug(Bug oBug)
        {
            return oBugDao.update(oBug);
        }

        public bool crearBug(Bug oBug)
        {
            return oBugDao.createBugConHistorial(oBug);
        }
             
}
  }

File: Clases 2018/Clases-C#/Simulacro_parcial_2_2018 (creacion de bug e historico en transaccion)/VB_TPS_BugTracker/TPS_BugTracker/BusinessLayer/Services/BugService.vb
Public Class BugService
    Private oBugDao As BugDao

    Public Sub New()
        oBugDao = New BugDao()
    End Sub

    Public Function consultarBugsConFiltros(ByVal params As List(Of Object)) As IList(Of Bug)
        Return oBugDao.getBugByFilters(params)
    End Function

    Public Function consultarBugsPorId(ByVal id As Integer) As Bug
        Return oBugDao.getBugById(id)
    End Function

    Public Function actualizarBug(ByVal oBug As Bug) As Boolean
        Return oBugDao.update(oBug)
    End Function

    Public Function crearBug(ByVal oBug As Bug) As Boolean
        Return oBugDao.createBugConHistorial(oBug)
    End Function
End Class